pkg/controllers/uiplugin/components.go

```diff
@@ -252,10 +268,172 @@ func newService(info UIPluginInfo, namespace string) *corev1.Service {
+		{
+			Name:      servingCertVolumeName,
+			ReadOnly:  true,
+			MountPath: "/secrets/",
+		},
+	}
+
+	volumes = append(volumes, corev1.Volume{
+		Name: "korrel8r-config",
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: name,
+				},
+			},
+		},
+	})
+	volumeMounts = append(volumeMounts, corev1.VolumeMount{
+		Name:      "korrel8r-config",
+		ReadOnly:  true,
+		MountPath: Korrel8rConfigMountDir,
+	})
+
+
+	deploy := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: appsv1.SchemeGroupVersion.String(),
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    componentLabels(name),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: componentLabels(name),
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+					Labels:    componentLabels(name),
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: info.Name + serviceAccountSuffix,
+					Containers: []corev1.Container{
+						{
+							Name:    name,
+							Image:   info.Korrel8rImage,
+							Command: []string{"korrel8r", "web", "--https=:8443", "--cert=/secrets/tls.crt", "--key=/secrets/tls.key", "--config=/config/korrel8r.yaml"},
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 8443,
+									Protocol:      corev1.ProtocolTCP,
+								},
+							},
+							SecurityContext: &corev1.SecurityContext{
+								RunAsNonRoot:             ptr.To(true),
+								AllowPrivilegeEscalation: ptr.To(false),
+								Capabilities: &corev1.Capabilities{
+									Drop: []corev1.Capability{
+										"ALL",
+									},
+								},
+								SeccompProfile: &corev1.SeccompProfile{
+									Type: corev1.SeccompProfileTypeRuntimeDefault,
+								},
+							},
+							VolumeMounts: volumeMounts,
+						},
+					},
+					Volumes: volumes,
+				},
+			},
+		},
+	}
+	return deploy
+}
+
+func newKorrel8rService(name string, namespace string) *corev1.Service {
+	annotations := map[string]string{
+		"service.alpha.openshift.io/serving-cert-secret-name": name,
+	}
+
+	return &corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "Service",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        name,
+			Namespace:   namespace,
+			Labels:      componentLabels(name),
+			Annotations: annotations,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Port:       port,
+					Name:       "web",
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr.FromInt32(port),
+				},
+			},
+			Selector: componentLabels(name),
+			Type:     corev1.ServiceTypeClusterIP,
+		},
+	}
+}
+
+func newKorrel8rConfigMap(name string, namespace string) *corev1.ConfigMap {
+
+	korrel8rData := struct {
```

Comment:
This is fine but just FYI, a map is also a handy way to feed data to a template:
```
korrel8rData := map[string]string { "Metric": "thanos-querier", ... etc }
```

Reply:
Done